### PR TITLE
[APM] rename trace_root type to trilean

### PR DESCRIPTION
**Pull request by simont1**
### What does this PR do?
In the backend, we renamed the field that `is_trace_root` uses from `TraceRootFlag` to `Trilean` since it can only have three possible values: `TRUE, FALSE, NOT_SET`. The purpose of this PR is to maintain consistency in the naming between the agent payload field and the backend payload field

### Motivation

The `Trilean` field name is better for shared code purposes. We might want to create other variables that implement this type

### Possible Drawbacks / Trade-offs

N/A

### Describe how to test/QA your changes

This is not a new feature, no new tests need to be created. I have changed `is_trace_root` field's existing e2e tests to use the new enum name.

### CODE REVIEW

**Review by kitfre — APPROVED** [review]

**Review by mx-psi — COMMENTED** [review]
cc @DataDog/opentelemetry this is a breaking change, do we want to add back the original symbols as deprecated until we migrate?

**Review by urseberry — APPROVED** [review]

File: pkg/proto/datadog/trace/stats.proto [hunk 1]
```diff
@@ -62,7 +62,7 @@ message ClientStatsBucket {
 	int64 agentTimeShift = 4;
 }
 
-enum TraceRootFlag {
+enum Trilean {
```
Comment (ajgajg1134) [inline pkg/proto/datadog/trace/stats.proto:65]: I had to look up the name `Trilean` as I was totally unfamiliar with this term, is this a term that already exists in the backend or in other areas? If not perhaps a better name might be something like OptionalBoolean?

Comment (simont1) [reply pkg/proto/datadog/trace/stats.proto:65]: We're already using `Trilean` in the backend

**Review by mx-psi — APPROVED** [review]

**Review by ajgajg1134 — APPROVED** [review]

**Pull request merged**